### PR TITLE
Add Ganon Boss Key Condition: Tokens

### PR DESCRIPTION
**Pull request by Cuphat**
For consistency with bridge conditions and more flexibility, adds 'On LACS: Tokens' as a Ganon BK setting to give the BK at a variable number of Gold Skulltula Tokens.

Also fixes a few minor errors in the calculation of stones/medallions required by settings and the use of the wrong setting name which caused LACS: Dungeons to fail to generate.

### CODE REVIEW

File: SettingsList.py [hunk 1]
```diff
@@ -2787,19 +2788,21 @@ def __init__(self, name, gui_text, min, max, default, step=1,
             'On LACS: Vanilla': Shadow and Spirit Medallions.
             'On LACS: Medallions': A configurable amount of Medallions.
             'On LACS: Stones': A configurable amount of Spiritual Stones.
-            'On LACS: Dungeons': A configurable amount of Dungeon Rewards
+            'On LACS: Dungeons': A configurable amount of Dungeon Rewards.
+            'On LACS: Tokens': A configurable amount of Gold Skulltula Tokens.
                                  (Spiritual Stones & Medallions).
```
Comment (Roman971) [inline SettingsList.py]: That 2nd line is part of the dungeons option. I don't really like having it tbh, but I feel like "dungeon rewards" might be unclear and leaving it on the same line is too long. So not sure how it should be.
```suggestion
            'On LACS: Dungeons': A configurable amount of Dungeon Rewards
                                 (Spiritual Stones & Medallions).
            'On LACS: Tokens': A configurable amount of Gold Skulltula Tokens.
```

File: Hints.py [hunk 2]
```diff
@@ -896,7 +896,9 @@ def buildGanonBossKeyString(world):
             elif world.lacs_condition == 'stones':
                 item_req_string = str(world.lacs_stones) + ' ' + item_req_string
             elif world.lacs_condition == 'dungeons':
-                item_req_string = str(world.lacs_dungeons) + ' ' + item_req_string
+                item_req_string = str(world.lacs_rewards) + ' ' + item_req_string
```
Comment (Roman971) [inline Hints.py:899]: Nice catch. I guess bridge has the same issue?

File: SettingsList.py (hunk 1 above)
Comment (Cuphat) [reply SettingsList.py]: Not sure how I missed that second line when adding in tokens. Whoops.

File: Hints.py (hunk 2 above)
Comment (Cuphat) [reply Hints.py:899]: I does, but apparently I missed it while being hyper focused on getting the Ganon BK for tokens to work. Fixed that as well.

**Comment by Cuphat** [comment]
Fixed the dungeons bridge and the whoopsie in the settings description and force pushed. Thanks for pointing those out.

**Comment by Roman971** [comment]
Thanks for the token addition and fixes.
I hope you could appreciate my great asm code for bridge/lacs triggers 👌

**Pull request merged**